spec/spec.md

```diff
@@ -631,36 +631,36 @@ where:
 
   - `transaction_valid`: an array of booleans with the same length as `payloads` indicating which are valid
 - `ERROR_NOT_SIGNED` if the wallet endpoint declined to sign this transaction for any reason
-- `ERROR_NOT_COMMITTED`
+- `ERROR_NOT_SUBMITTED`
 
   ```
   “data”: {
       “signatures”: [“<transaction_signature>”, ...],
-      “commitment”: [<commitment_reached>, ...],
+      “submitted”: [<submitted>, ...],
   }
   ```
 
-  if the requested commitment level was not reached for any of the signed transactions, where:
+  if the wallet endpoint was unable to submit one or more of the signed transactions to the network, where:
 
   - `signatures`: as defined for a successful result
-  - `commitment_reached`: for each entry in `signatures`, a boolean indicating whether the desired commitment level was reached
+  - `submitted`: for each entry in `signatures`, a boolean indicating whether the transaction was submitted to the network
```

Comment:
Feels like we should define some of this a bit more clearly --

> a successful result

What does this mean? If the signature / txid was returned by the RPC node, is that a successful result?

> submitted to the network

What does this mean? Why would a transaction _not_ be submitted to the network? Does it mean an RPC node didn't respond with the txid?

Reply:
> > a successful result
> 
> What does this mean? If the signature / txid was returned by the RPC node, is that a successful result?

Basically, if [`sendTransaction`](https://docs.solana.com/developing/clients/jsonrpc-api#sendtransaction) returns success. I'll make the text more explicit on what a success is.

> > submitted to the network
> 
> What does this mean? Why would a transaction not be submitted to the network? Does it mean an RPC node didn't respond with the txid?

Same answer - did an RPC node accept the transaction. I'll update the text to better capture the low bar of success for this method call.

Reply:
Hmm. So in the case that sending failed, the wallet should still respond with signatures, which the dapp could then send itself? This sounds like a potential security issue, since the `signAndSendTransaction` flow is generally preferred by wallets to avoid exposing the signature _unless_ the wallet does the sending.

Reply:
Could responding with `{ signatures: [<signature>, null, <signature>] }` instead be an option? I'm not sure how useful or reliable the information from `submitted` is, because a dapp has to check to see if the tx was processed/confirmed/finalized anyway. If it wasn't submitted, I would think the dapp should request the wallet to sign and send it again rather than send itself. For example, it may be a metatransaction anyway, and therefore the dapp can't just send itself.

Reply:
Yep, that sounds reasonable - a null entry in the array conveys the failure of sending a particular transaction in the set, and avoids exposing a signature that a dapp could attempt to use itself.